### ISSUE TITLE
Is the latest version of DALI compatible with Tensorflow v1.13 and 1.14?

### ISSUE THREAD

**Issue by KleinXin** [NVIDIA/DALI#1871]
I am using Tensorflow v1.13 and 1.14.
Which version of DALI is recomended?

**Comment by KleinXin** [comment]
In the developer guide of DALI v0.18.0, a tutorial of multi-gpus is provided
[https://docs.nvidia.com/deeplearning/sdk/dali-archived/dali_0180_beta/dali-developer-guide/docs/examples/multigpu.html](url)

A sample code of Tensorflow v1.10.0 is also provided
[https://docs.nvidia.com/deeplearning/sdk/dali-archived/dali_0180_beta/dali-developer-guide/docs/examples/tensorflow/demo/README.html](url)

So I guess maybe DALI v0.18.0 is compatible with Tensorflow v1.13 and v1.14

I will try to use DALI v0.18.0.

Any suggestion is appreciated!

**Comment by JanuszL** [comment]
Hi,
Up to version 0.21 (should be released soon) DALI supports TF 1.14. Starting from 0.22 0.15.x will be the only CUDA 10 supported version so I recommend using 0.20 for now.

**Comment by KleinXin** [comment]
> Hi,
> Up to version 0.21 (should be released soon) DALI supports TF 1.14. Starting from 0.22 0.15.x will be the only CUDA 10 supported version so I recommend using 0.20 for now.

Thank you for your reply!
I am now trying to use DALI v0.18.0.

I am trying to create the pipeline following the sample code below. 

`from nvidia.dali.pipeline import Pipeline
import nvidia.dali.ops as ops
import nvidia.dali.types as types
import nvidia.dali.tfrecord as tfrec
import numpy as np
import matplotlib.pyplot as plt


class TFRecordPipeline(Pipeline):
    def __init__(self, batch_size, num_threads, device_id):
        super(TFRecordPipeline, self).__init__(batch_size,
                                         num_threads,
                                         device_id)
        self.input = ops.TFRecordReader(path = tfrecord,
                                        index_path = tfrecord_idx,
                                        features = {"image/encoded" : tfrec.FixedLenFeature((), tfrec.string, ""),
                                         'image/class/label':         tfrec.FixedLenFeature([1], tfrec.int64,  -1),
                                         'image/class/text':          tfrec.FixedLenFeature([ ], tfrec.string, ''),
                                         'image/object/bbox/xmin':    tfrec.VarLenFeature(tfrec.float32, 0.0),
                                         'image/object/bbox/ymin':    tfrec.VarLenFeature(tfrec.float32, 0.0),
                                         'image/object/bbox/xmax':    tfrec.VarLenFeature(tfrec.float32, 0.0),
                                         'image/object/bbox/ymax':    tfrec.VarLenFeature(tfrec.float32, 0.0)})
        self.decode = ops.ImageDecoder(device = "mixed", output_type = types.RGB)
        self.resize = ops.Resize(device = "gpu", resize_shorter = 256.)
        self.cmnp = ops.CropMirrorNormalize(device = "gpu",
                                            output_dtype = types.FLOAT,
                                            crop = (224, 224),
                                            image_type = types.RGB,
                                            mean = [0., 0., 0.],
                                            std = [1., 1., 1.])
        self.uniform = ops.Uniform(range = (0.0, 1.0))
        self.iter = 0

    def define_graph(self):
        inputs = self.input()
        images = self.decode(inputs["image/encoded"])
        resized_images = self.resize(images)
        output = self.cmnp(resized_images, crop_pos_x = self.uniform(),
                           crop_pos_y = self.uniform())
        return (output, inputs["image/class/text"])

    def iter_setup(self):
        pass
`
However, it needs an index file "tfrecord_idx". What is the format of this file?

When I tried to generate this file by following the sample code below, it gave a permission denied error. Do I need to have the root permission?

`from subprocess import call
import os.path

test_data_root = os.environ['DALI_EXTRA_PATH']
tfrecord = os.path.join(test_data_root, 'db', 'tfrecord', 'train')
tfrecord_idx = "idx_files/train.idx"
tfrecord2idx_script = "tfrecord2idx"

if not os.path.exists("idx_files"):
    os.mkdir("idx_files")

if not os.path.isfile(tfrecord_idx):
    call([tfrecord2idx_script, tfrecord, tfrecord_idx])`

![image](https://user-images.githubusercontent.com/13084622/79290785-92a0e680-7eff-11ea-9ac0-93fe1704cb13.png)

All the sample codes I used can be found on the page of developer guide of DALI v0.18.0

[https://docs.nvidia.com/deeplearning/sdk/dali-archived/dali_0180_beta/dali-developer-guide/docs/examples/dataloading_tfrecord.html](url)

**Comment by JanuszL** [comment]
Hi,
Make sure you can execute `tfrecord2idx` script itself and if the destination path is writing, in this case, `idx_filestrain.idx`.
Also try to run this command from cmd (not from python layer)
